### ISSUE TITLE
Iteration 1 DEBRIEF

### ISSUE THREAD

**Issue by dominictarr** [dominictarr/feedopensource#31]
Okay!

I'm gonna declare ITERATION 1 COMPLETE.

It's not perfect, but it's amazingly more featureful than when you all funded it.

Thanks to everyone who contributed to the first iteration!

@ralphtheninja @NHQ @binarykitchen @substack @bluntworks @gotcha @julianduque @hansent @levity @eugeneware @eins78 @henrock  @joates  @grncdr @rmetzler

Now, I have something more to ask of you all - have a look at what I've done and give me feedback!

Especially: _Run your own feedopensource project_ see [docs](https://github.com/dominictarr/feedopensource/blob/master/docs.md)

Or maybe set up your own instance: [development](https://github.com/dominictarr/feedopensource/blob/master/development.md) (not necessary, but just in case you want to)

I have a few things in mind for the next step - but what I need more is a outsider perspective on whats working well, and what needs improvement, and what the biggest pain points are when you try to use feedopensource your self.

**Comment by Raynos** [comment]
@dominictarr can you demonstrate the iteration status badge ? Maybe place a section on the README of "iterations" with the status badges ?

This uri https://feedopensource.com/iteration/dominictarr/feedopensource/1PTAwipYpP63uNrcxfm5FewxRdZyar6ceu.png#1.2 doesn't do anything for me. Doesn't render an image.

Can you document or talk about how payment is handled, i.e. how do I know which github users paid ? Can I see in the API somewhere a breakdown of that information ?

**Comment by Raynos** [comment]
@dominictarr how did you fetch feedback on what should be in "iteration 1" before you started iteration 1. I feel like I would want to get suggestions from backers on what should be in "iteration 1" rather then deciding it myself.

**Comment by Raynos** [comment]
@dominictarr to confirm. Is there a single feedopensource badge ? that changes to "progress" mode once the funding is completed ? Or are there two badges ? one progress bar for funding and one progress for tasks ?

**Comment by dominictarr** [comment]
ah... I thought I had fixed the github rate limit but I'm still seeing 403s. there is something still wrong.

**Comment by dominictarr** [comment]
okay - it was two problems combined - one was a missing `&` wasn't separating two query parameters,
and the other one was a config path thing. when I ran the app with systemctl there is no `$HOME` and so it didn't find the config file with auth, and so github gave it a 403. This all fixed now.

**Comment by Raynos** [comment]
It would be nice if there were a set of images of the feedopensource badge in multiple stages (created, funding, progress, complete)

**Comment by binarykitchen** [comment]
@dominictarr Thanks so much! I will try https://github.com/dominictarr/feedopensource/blob/master/docs.md in the next weeks when I have a mad project ready ...

Just a small thing in that docs.md: What is the difference between a bitcoin wallet and a bitcoin address? Once you mention wallet, then in the badge example address.

**Comment by eins78** [comment]
> What is the difference between a bitcoin wallet and a bitcoin address? 

An address is an identifier with has a certain spendable amount (bank terms: savings account). 
A wallet is a collection of between 1 and lots of addresses (bank term: portfolio).

So end users deal with wallets, which can contain several addresses. From the outside view (blockchain) there is no way to know which addresses are together in a wallet; also addresses can be in more than one wallet. 

TL;DR: you can open a wallet on blockchain.info/wallet and have as many addresses as you want in it. 

> Once you mention wallet, then in the badge example address.

So in this context only **addresses** should be mentioned (not submitting PR because I should not commit from my phone).

**Comment by binarykitchen** [comment]
@eins78 I see, thanks.

@dominictarr I think this sentence in your docs.md ...

_each iteration MUST use a new bitcoin wallet_

 ... is meant to be ...

_each iteration MUST use a new bitcoin address_

**Comment by eins78** [comment]
Now that you mentioned it I spotted one more error, fixed in https://github.com/dominictarr/feedopensource/pull/36

I think the best way to further reduce confusion would be a screenshot of a blockchain.info wallet, with a few addresses which are labeled like: "my project - iteration 1", "… iteration 2", "unrelated beer money", etc.

**Comment by joates** [comment]
(edit _22-Jan-2014 **this is now resolved**_ -> [commit](https://github.com/joates/feedopensource/commit/914d1f18dab5d2683fba8bc10f8cc39e390006e3) )

started testing with this [issue](https://github.com/xtcrypto/funding-project/issues/1) .. ~~_the badge isn't working_~~ (edit _hyphen in the repo name is a separate issue_ -> #43 )
- there were some hard-coded https requests to `feedopensource.com` which i had to change in `./api/iterations.js`  
- i get this [error](https://xtco.in/iteration/xtcrypto/funding-project/1d786d66-f357-4ef5-8165-1772ee85e673) using the frontend api.  
- this doesn't work either `node ./api/index.js iteration dominictarr feedopensource 1PTAwipYpP63uNrcxfm5FewxRdZyar6ceu` :(  

screenshot: (showing output from process and querying the cli api)
![xtcoin-iteration-not-found-error_screenshot](https://f.cloud.github.com/assets/490654/1955722/d0cf56e2-81ec-11e3-8c6e-4d29712e0116.png)

indeed there are no iteration(s) reported ?? ( _returned empty array_ ) !!

probably something simple/basic i'm missing.. 
**debugging tips appreciated, thanks**

**Comment by joates** [comment]
(edit _22-Jan-2014 **this is now resolved**_ -> [commit](https://github.com/joates/feedopensource/commit/d1850c1b0c6a2ae1539fa7edecd3f09959b1349f) )

ok, i got it to work.

here is the patch.. (look for **change_this**)

``` javascript
diff --git a/api/iterations.js b/api/iterations.js
index d680597..d8d69f2 100644
--- a/api/iterations.js
+++ b/api/iterations.js
@@ -3,9 +3,8 @@ var join    = require('path').join

 var regex   = /^iteration/i  // starts with 'Iteration'

-//var progressbar      = /https?:\/\/feedopensource\.com\/(?:badge|iteration\/\w+\/w+)\/(1[1-km-z]{33})\/(\d+(?:\.\d+))/
 var progressbar      =
-  /https?:\/\/feedopensource\.com\/iteration\/[0-z_]+\/[0-z_]+\/(1[1-km-z]{33}).png#(\d+(?:\.\d+))/
+  /https?:\/\/change_this\.com\/iteration\/[0-z_]+\/[0-z_]+\/(1[1-km-z]{33}).png#(\d+(?:\.\d+))/
 var issueFull        = /https?:\/\/github.com\/([0-z_]+)\/([0-z_]+)\/issues\/(\d+)/
 var issueNum         = /\s#(\d+)\s/
 var issueUserNum     = /\s([0-z_]+)#(\d+)\s/
@@ -51,7 +50,7 @@ module.exports = function (issues, collaborators, user, repo) {
     if(!m) return
     issue.wallet = m[1]
     issue.target = m[2]
-    issue.feedopensource_url = 'https://feedopensource.com/iteration/' + user + '/' + repo + '/' + m[1]
+    issue.feedopensource_url = 'https://change_this.com/iteration/' + user + '/' + repo + '/' + m[1]
     issue.badge_url = issue.feedopensource_url + '.png#' + issue.target

     if(!wallets[issue.wallet]) {
```

**Comment by joates** [comment]
also, **worth noting**
to access your wallet on blockchain.info you use a wallet _identifier_ and your password.

> The bitcoin wallet's _identifier_ is NOT the same as a bitcoin address !!

i was using the identifier by mistake :(

**Comment by joates** [comment]
@dominictarr having to dig into the source files to change the domain was a pain point, as you commented in irc chat this would be improved if it was included with the other settings in config.js.

**Comment by joates** [comment]
> I'm gonna declare ITERATION 1 COMPLETE.

@dominictarr >> at what point do we stop bug squashing? and say..
"ok from this point on we are tasking for [iteration #2](https://github.com/dominictarr/feedopensource/issues/20)"

since i'm not a certified scrum master i don't really have an opinion, just asking because at some point a decision needs to be made.

just a final thought, it's a learning process (for me especially), i guess the project will evolve at it's own [pace](https://github.com/dominictarr/feedopensource/issues/28), might be useful to document some guidelines as we go along ?

(edit) _perhaps there exists an opensource calender app that we can leverage so that a period of downtime between sprints can be scheduled and @dominictarr can have some space to work on other projects and then return to this in 7-10-15 days or whatever._

**Comment by dominictarr** [comment]
configuration in the source code is not desirable.
instead use `new Regexp(string1 + string1 + ...)`
